core/src/main/scala/kafka/log/Log.scala

```diff
@@ -217,6 +218,9 @@ class Log(@volatile var dir: File,
         if (indexFileExists) {
           try {
             segment.index.sanityCheck()
+            // Resize the time index file to 0 if it is newly created.
+            if (!timeIndexFileExists)
+              segment.timeIndex.resize(0)
```

Comment:
The offset/time index is trimmed when the offset/time index is closed. We already trim the offset/time index in LogSegment.recover().